src/Contracts/OpdexMiningGovernance.cs

```diff
@@ -249,17 +251,15 @@ private void SetMiningPoolRewardAmountExecute()
         Notified = false;
     }
     
-    private Address Deploy(Address stakingPool)
+    private Address FindMiningPool(Address stakingPool)
     {
         var miningPool = GetMiningPool(stakingPool);
 
         if (miningPool != Address.Zero) return miningPool;
 
-        miningPool = Create<OpdexMiningPool>(0ul, new object[] { Address, MinedToken, stakingPool, MiningDuration }).NewContractAddress;
+        miningPool = (Address)Call(stakingPool, 0ul, "get_MiningPool").ReturnValue;
```

Comment:
If a staking pool without a mining pool (`stakingToken == token`) gets nominated then could it result in nominations forever being locked? I believe `RewardMiningPool` would always fail and so nominations could never get reset. Not tested this.

Reply:
That can't happen because `EnsureStakingEnabled` will fail for the staking pool that doesn't have a mining where `stakingToken == token`.  So nominations wouldn't ever be possible in this scenario. Good callout though 👍  that's not easily visible when browsing this PR.

Reply:
Edit: Actually, you're right and I should try to be protective here somehow, good catch.

Techincally, any third party contract could "stake", nominations call `MinedToken` then that calls `MiningGovernance` which is where this would come into play. Any potentially nominated third party contract that didn't implement an `IOpdexStakingPool` interface could lock everything.

Reply:
This can't happen because the `NominateLiquidityPool` method would fail which would mean so did the staking transaction that triggered it. 

In staking pools, we only nominate eligable staking pools with `EnsureStakingEnabled()`. https://github.com/Opdex/opdex-v1-core/blob/feature/SC-1_MigrateMiningPool/src/Contracts/Pools/OpdexStakingPool.cs#L93-L96

I will add a check for Address.Zero though and fail on that result when retrieving a mining pool.

Reply:
Rethinking this, you're right...

In the case that the `MiningPool` associated with a `StakingPool` isn't using the correct interface `NotifyRewardAmount` in the mining pool contact could fail and lock up the process. Will have to think about this one...

Might require a test call during nomination to `NoftifyRewardAmount` with a 0 amount, requiring a successful response.

Reply:
This is resolved in 64a1b3c no longer check for successful `NotifyRewardAmount` preventing any possible lockup. During nominations, we also validate the mining pool addresses that are looked up.